### ISSUE TITLE
Pass <Component/> to `label`

### ISSUE THREAD

**Issue by pudgereyem** [mathieudutour/react-swipe-to-reveal-options#10]
Problem:
I wan't to render a component in the "label". Something like below:

```
var items = [
          {
            leftOptions: [{
              label: <Icon type="delete"/>, <--
              class: 'trash'
            }],
           ...
]
```

I looked into the code [https://github.com/enkidevs/react-swipe-to-reveal-options/blob/master/react-swipe-to-reveal-options.js](https://github.com/enkidevs/react-swipe-to-reveal-options/blob/master/react-swipe-to-reveal-options.js) and I saw that you accept `html` by using `dangerouslySetInnerHTML`.

I would be more than happy to fix so that it accepts components aswell. Should I change the above file directly or do you build it out from any source code? Can't find anything in the repo. And  [https://github.com/enkidevs/react-swipe-to-reveal-options/blob/master/react-swipe-to-reveal-options.js](https://github.com/enkidevs/react-swipe-to-reveal-options/blob/master/react-swipe-to-reveal-options.js) was quite "bloated".

Thanks.

**Comment by mathieudutour** [comment]
Yeah actually just removing `dangerouslySetInnerHTML` should work right? I don't recall why it was there in the first place.

Happy to accept a PR. I can look into it tomorrow maybe otherwise.
It's not built from anything, just raw js.

**Comment by avk** [comment]
+1 I actually ran into just this problem myself yesterday.

**Comment by mathieudutour** [comment]
Should be fixed with v0.3.0

**Comment by avk** [comment]
v0.3.0 seems to work with React components but plain old HTML strings now no longer work. I don't mind supporting only React components as labels, since they're far more flexible, and making people write JSX (or what it compiles to). 

Alternatively, you could branch based on the type of the label. If it's a string or DOM node, do the old `dangerouslySetInnerHTML` method, if it's anything else, treat it as you do in v0.3.0 and let the wrapper just try to render it.

I don't have a preference. Though whatever this change ends up being, it should be updated in the README and examples.

Thoughts?

**Comment by mathieudutour** [comment]
Yeah definitely. Should be fixed in 0.3.1 then :)

It's becoming really messing, I should rewrite it with babel...

**Comment by darrenmcgrath** [comment]
Hello, just wondering if this was ever implemented as I am struggling to get html into the main content area? I am successfully getting html into both the left and right options but is is still rendering the main content as a string.
Cheers!
